### ISSUE TITLE
TODO labels are missing in Spyder 4

### ISSUE THREAD

**Issue by jondo** [spyder-ide/spyder#7963]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports

## Problem Description
There are no TODO labels dispayed any more in the editor margin of Spyder 4.

### What steps reproduce the problem?

1. Start Spyder 4 from the current master branch.
2. Enter a `#TODO` line in the editor.
3. Examine the editor margin left ofthe line.

### What is the expected output? What do you see instead?
A TODO label should appear, as in Spyder 3.x.

### Remarks
@ccordoba12 already [confirmed](https://github.com/spyder-ide/spyder/issues/2264#issuecomment-425569597) the problem.

Just tell me if you need to know more about my test environment.

**Comment by CAM-Gerlach** [comment]
Yup, I've confirmed on my Win 8.1/Py3.6/Qt5.9 machine running the latest Spyder ``master`` that this affects all of the comments (``XXX``, ``FIXME``, etc) previously labelled with a check mark, while this is not the case for the latest ``3.x``.

@jondo can you run ``git bisect`` or similar to see when the regression was introduced?

**Comment by jondo** [comment]
Testing with the file
```
#TODO foo
#XXX bar
#FIXME baz
```

gives me the following, where all three lines stop working:

```
1db53b279c18e673e567559da2e71d71f4a6b8d8 is the first bad commit
commit 1db53b279c18e673e567559da2e71d71f4a6b8d8
Author: Edgar Andrés Margffoy Tuay <andfoy@gmail.com>
Date:   Wed Oct 4 18:15:21 2017 -0500

    Improve socket reading procedure
```

Can that be correct?

**Comment by CAM-Gerlach** [comment]
Thanks! Possibly. but I'm not sure exactly what it would be so... @ccordoba12 ? Looks like [the relevant commit](https://github.com/spyder-ide/spyder/commit/1db53b279c18e673e567559da2e71d71f4a6b8d8) was merged in with the LSP changes, so that could explain why you might not have noticed it until recently. I wasn't able to confirm it testing myself, since I got multiple errors just trying to run that version, likely due to dependency versions it wasn't designed to be compatible with (Qt 5.9, etc).
